### PR TITLE
feat: csv handling param

### PR DESCRIPTION
**Pull request by hansmrtn**

### CODE REVIEW

File: src/client/baseClient.ts [hunk 1]
```diff
@@ -35,7 +37,11 @@ abstract class BaseClient {
           );
         }
 
-        return resp.json();
+        if (csv) {
```
Comment (gravityblast) [inline src/client/baseClient.ts]: @hmrtn instead of having a csv parameter we can parse the body based on the content type in the response header

**Review by gravityblast — CHANGES_REQUESTED** [review]

**Comment by hansmrtn** [comment]
Reopened this in reference to: https://github.com/gitcoinco/allo-indexer/pull/56 in regards to the ticket: https://github.com/gitcoinco/grants-stack/issues/1475

**Comment by vacekj** [comment]
https://github.com/gitcoinco/allo-indexer/pull/56#discussion_r1182348568

**Comment by hansmrtn** [comment]
okie

**Pull request closed without merging**